lib/streamlit/elements/progress.py

```diff
@@ -29,6 +30,27 @@
 FloatOrInt: TypeAlias = Union[int, float]
 
 
+def _check_float_between(value: float, low: float = 0.0, high: float = 1.0) -> bool:
+    """
+    Checks given value is 'between' the bounds of [low, high],
+    considering close values around bounds are acceptable input
+
+    Parameters
+    ----------
+    value : float
+    low : float
+    high : float
+    """
+    if (
+        (low <= value <= high)
+        or math.isclose(value, low, rel_tol=1e-9, abs_tol=1e-9)
+        or math.isclose(value, high, rel_tol=1e-9, abs_tol=1e-9)
+    ):
+        return True
+    else:
+        return False
```

Comment:
nit: this could be cleaned up a tiny bit by just returning the conditional in the `if` rather than returning `True` or `False` based on it

```python3
    return (
        (low <= value <= high)
        or math.isclose(value, low, rel_tol=1e-9, abs_tol=1e-9)
        or math.isclose(value, high, rel_tol=1e-9, abs_tol=1e-9)
    )
```

Reply:
@vdonato I thought at first since the if condition is quite long it would hurt readability to simply return the expression, but now that I see your suggestion it looks cleaner. Thank you for the comment. I have fixed it in commit [98ff93b4b3af986ef4fde89b8689a22c81dc7a09](https://github.com/streamlit/streamlit/pull/7953/commits/98ff93b4b3af986ef4fde89b8689a22c81dc7a09).